Source/ASDisplayNode+Layout.mm

```diff
@@ -536,17 +543,28 @@ - (void)transitionLayoutWithSizeRange:(ASSizeRange)constrainedSize
     return;
   }
     
-  // Check if we are a subnode in a layout transition.
-  // In this case no measurement is needed as we're part of the layout transition.
-  if ([self _isLayoutTransitionInvalid]) {
-    return;
-  }
-  
+  BOOL shouldInvalidateLayout = NO;
   {
     ASDN::MutexLocker l(__instanceLock__);
-    ASDisplayNodeAssert(ASHierarchyStateIncludesLayoutPending(_hierarchyState) == NO, @"Can't start a transition when one of the supernodes is performing one.");
+
+    // Check if we are a subnode in a layout transition.
+    // In this case no measurement is needed as we're part of the layout transition.
+    if ([self _locked_isLayoutTransitionValid]) {
+      return;
+    }
+
+    if (ASHierarchyStateIncludesLayoutPending(_hierarchyState)) {
```

Comment:
Can there be any problems if we call ASHierarchyStateIncludesLayoutPending while holding the lock?